lib/adal.js

```diff
@@ -1082,29 +1082,43 @@ var AuthenticationContext = (function () {
         // Format: xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx
         // y could be 1000, 1001, 1010, 1011 since most significant two bits needs to be 10
         // y values are 8, 9, A, B
-        var guidHolder = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
-        var hex = '0123456789abcdef';
-        var r = 0;
-        var guidResponse = "";
-        for (var i = 0; i < 36; i++) {
-            if (guidHolder[i] !== '-' && guidHolder[i] !== '4') {
-                // each x and y needs to be random
-                r = Math.random() * 16 | 0;
+        var cryptoObj = window.crypto || window.msCrypto; // for IE 11
+        if (cryptoObj && cryptoObj.getRandomValues) {
+            var buffer = new Uint16Array(8);
+            cryptoObj.getRandomValues(buffer);
+            function decimalToHex(num) {
+                var hex = num.toString(16);
+                while (hex.length < 4) {
+                    hex = '0' + hex;
+                }
+                return hex;
             }
-
-            if (guidHolder[i] === 'x') {
-                guidResponse += hex[r];
-            } else if (guidHolder[i] === 'y') {
-                // clock-seq-and-reserved first hex is filtered and remaining hex values are random
-                r &= 0x3; // bit and with 0011 to set pos 2 to zero ?0??
-                r |= 0x8; // set pos 3 to 1 as 1???
-                guidResponse += hex[r];
-            } else {
-                guidResponse += guidHolder[i];
+            return decimalToHex(buffer[0]) + decimalToHex(buffer[1]) + '-' + decimalToHex(buffer[2]) + '-' + decimalToHex(buffer[3]) + '-' +
+              decimalToHex(buffer[4]) + '-' + decimalToHex(buffer[5]) + decimalToHex(buffer[6]) + decimalToHex(buffer[7]);
```

Comment:
This is not in the format: xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx. Can you check on the significance of '4' and 'y'. It seems to be related to rfc 4122